File: Documentation/ModeDevelopment/Reference Files/hl/highlight.pl
#!/usr/bin/perl -w
# This is a pseudo Perl file to test Kate's Perl syntax highlighting.
# TODO: this is incomplete, add more syntax examples!

sub prg($)
{
	my $var = shift;

	$var =~ s/bla/foo/igs;
	$var =~ s!bla!foo!igs;
	$var =~ s#bla#foo#igs;
	$var =~ tr/a-z/A-Z/;
	($match) = ($var =~ m/(.*?)/igs);

	$test = 2/453453.21;
	$test /= 2;

	print qq~d fsd fsdf sdfl sd~
	
	$" = '/';
	
	$foo = <<__EOF;
d ahfdklf klsdfl sdf sd
  SHOULD NOT MATCH: __EOF <--
fsd sdf sdfsdlkf sd
__EOF

	$x = "dasds";

	next if( $match eq "two" );
	next if( $match =~ /go/i );

	@array = (1,2,3);		# a comment
	@array = qw(apple foo bar);
	push(@array, 4);
	%hash = (red => 'rot',
		blue => 'blau');
	print keys(%hash);
}

sub blah {
}

&blah;
prg("test");

# highlighting '-f' as Operator.
$cfg = Config::IniFiles->new( -file => "/path/to/config_file.ini" );

File: Documentation/ModeDevelopment/Reference Files/hl/highlight.php
<?
/* This is a pseudo PHP file to test Kate's PHP syntax highlighting. */
# TODO: this is incomplete, add more syntax examples!
# this is also a comment.
// Even this is a comment
function test($varname) {
	return "bla";	# this is also a comment
}

?>

<?php echo("hello test"); ?>

<html>
	<? print "<title>test</title>"; ?>
</html>

<?php
$var = <<<DOOH
This is the $string inside the variable (which seems to be rendered as a string)
It works well, I think.
DOOH
?>

<?php

$test = <<<EOT

$foo bar

EOT

$test = <<<"EOT"

$foo bar

EOT

$test = <<<'EOT'

$foo bar

EOT

//html
$test = <<<EOHTML

<tag>$tag</tag>
<$tag>$foo</$tag>

EOHTML

$test = <<<"EOHTML"

<tag>$tag</tag>
<$tag>$foo</$tag>

EOHTML

$test = <<<'EOHTML'

<tag>$tag</tag>

EOHTML

//css
$test = <<<EOCSS

$class {
        $foo: $bar
}
.class {
        color: $color
}

EOCSS

$test = <<<"EOCSS"

$class {
        $foo: $bar
}
.class {
        color: $color
}

EOCSS

$test = <<<'EOCSS'

.class {
        color: $color
}

EOCSS

//javascript
$test = <<<EOJAVASCRIPT

var foo = "$foo";
function $bar() {
        $baz;
}

EOJAVASCRIPT

$test = <<<"EOJAVASCRIPT"

var foo = "$foo";
function $bar() {
        $baz;
}

EOJAVASCRIPT

$test = <<<'EOJAVASCRIPT'

var foo = "$foo";
function $bar() {
        $baz;
}

EOJAVASCRIPT

//mysql
$test = <<<EOMYSQL

SELECT $foo FROM $bar WHERE id = $baz;

EOMYSQL

$test = <<<"EOMYSQL"

SELECT $foo FROM $bar WHERE id = $baz;

EOMYSQL

$test = <<<'EOMYSQL'

SELECT $foo FROM $bar WHERE id = $baz;

EOMYSQL

//no EO

//html
$test = <<<HTML

<tag>$tag</tag>

HTML

$test = <<<"HTML"

<tag>$tag</tag>

HTML

$test = <<<'HTML'

<tag>$tag</tag>

HTML

//css
$test = <<<CSS

.class {
        color: $color
}

CSS

$test = <<<"CSS"

.class {
        color: $color
}

CSS

$test = <<<'CSS'

.class {
        color: $color
}

CSS

//javascript
$test = <<<JAVASCRIPT

var foo = "$foo";
function $bar() {
        $baz;
}

JAVASCRIPT

$test = <<<"JAVASCRIPT"

var foo = "$foo";
function $bar() {
        $baz;
}

JAVASCRIPT

$test = <<<'JAVASCRIPT'

var foo = "$foo";
function $bar() {
        $baz;
}

JAVASCRIPT

//mysql
$test = <<<MYSQL

SELECT $foo FROM $bar WHERE id = $baz;

MYSQL

$test = <<<"MYSQL"

SELECT $foo FROM $bar WHERE id = $baz;

MYSQL

$test = <<<'MYSQL'

SELECT $foo FROM $bar WHERE id = $baz;

MYSQL

?>